keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,577 @@
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Add a standard set of ObjectMeta fields (name, labels, and annotations) to the ComponentConfig API's.
+  - kubelet
+  - kube-proxy
+  - kube-scheduler
+  - kube-controller-manager
+  - kube-apiserver
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Graduate the API versions
+  - We will implement ObjectMeta within the current version of each ComponetConfig API
+- Implement every possible ObjectMeta field for ComponentConfig API's
+  - As these configs are singular per component, per cluster, they won't have or need the full suite of ObjectMeta similar to core Kubernetes API's (ie. Pods, Servcies, etc.)
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+- for component in [kubelet kubeproxy kubecontrollermanager kubeapiserver kubescheduler]
+  - Add the following ObjectMeta fields
+    - name
+    - labels
+    - annotations
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+- None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).  Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
```

Comment:
Add some details on how we'll test this, e.g. resue the existing infrastructure for round-trip testing of componentconfig APIs.

Reply:
I have some questions for clarification on this, will followup in WG Meetings

Reply:
Had a chat with @mtaufen a while ago and have updated details accordingly.